### PR TITLE
[OKTA-640287] Added home page banner component in dev docs

### PR DESCRIPTION
**Pull request by ishan-krishna-okta**
## Description:
- **What's changed?** Added support for home page banner component. Usage guidelines can be found [here](https://oktawiki.atlassian.net/wiki/spaces/DOC/pages/2920874189/Adding+a+banner+on+homepage+in+dev+docs) 
- **Is this PR related to a Monolith release?** No

### Resolves:

* [OKTA-640287](https://oktainc.atlassian.net/browse/OKTA-640287)

### CODE REVIEW

File: packages/@okta/vuepress-theme-prose/components/Hero.vue [hunk 1]
```diff
@@ -1,25 +1,74 @@
...
+         -->
+        <HomePageBanner
+          v-if="showBanner"
+          primary-cta-text="Primary button"
+          primary-cta-link="/docs/guides/"
+          primary-link-target="_self"
+          secondary-cta-text="Secondary button"
+          secondary-cta-link="https://www.yahoo.com"
+          @onPrimaryCtaClick="handlePrimaryCtaClick()"
+          @onSecondaryCtaClick="handleSecondaryCtaClick()"
```
Comment (emikhasyak-okta) [inline packages/@okta/vuepress-theme-prose/components/Hero.vue:37]: onClick handlers add a lot of complexity and I don't see a good use case for them. With the current approach user will be redirected to `primary-cta-link` anyways.

Can we stick with a simple approach? If someone asks for the onclick handler we probably will need to reconsider links implementation.

File: packages/@okta/vuepress-theme-prose/global-components/HomePageBanner.vue [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    <div class="home-page-banner--description">
+      <slot name="description" />
+    </div>
+
+    <div
+      v-if="primaryCtaText || secondaryCtaText"
+      class="home-page-banner--cta-container"
+    >
+      <a
+        v-if="primaryCtaLink"
```
Comment (emikhasyak-okta) [inline packages/@okta/vuepress-theme-prose/global-components/HomePageBanner.vue:15]: Let's also check here for `primaryCtaText`. If we remove link and leave text that will break UI
image

File: packages/@okta/vuepress-theme-prose/global-components/HomePageBanner.vue [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      v-if="primaryCtaText || secondaryCtaText"
+      class="home-page-banner--cta-container"
+    >
+      <a
+        v-if="primaryCtaLink"  
+        :href="primaryCtaLink"
+        :target="primaryLinkTarget ? primaryLinkTarget : '_blank'"
+      >
+        <button
+          v-if="primaryCtaText"
```
Comment (emikhasyak-okta) [inline packages/@okta/vuepress-theme-prose/global-components/HomePageBanner.vue:20]: No need to verify the button text here, once the previous comment is addressed.

**Review by emikhasyak-okta — COMMENTED** [review]
Overall looks very good!
Have two asks:
- update a bit component's visibility checks, so that UI is always consistent
- there are no onclick handlers in the original design, let's add them later when we have a real use case.

File: packages/@okta/vuepress-theme-prose/global-components/HomePageBanner.vue (hunk 2 above)
Comment (ishan-krishna-okta) [reply packages/@okta/vuepress-theme-prose/global-components/HomePageBanner.vue]: Nice catch.. done!

File: packages/@okta/vuepress-theme-prose/components/Hero.vue (hunk 1 above)
Comment (ishan-krishna-okta) [reply packages/@okta/vuepress-theme-prose/components/Hero.vue]: Yeah makes sense. I have removed the handlers for now.

File: packages/@okta/vuepress-theme-prose/components/Hero.vue [hunk 4]
```diff
@@ -1,25 +1,64 @@
...
+    </section>
+  </div>
 </template>
+
+<script>
+export default {
+  name: "Hero",
+  data() {
+    return {
+      showBanner: true
```
Comment (emikhasyak-okta) [inline packages/@okta/vuepress-theme-prose/components/Hero.vue:60]: `showBanner` has to be false

**Review by emikhasyak-okta — COMMENTED** [review]
Looks good to me! Please change showBanner flag to false and then we're good to go

File: packages/@okta/vuepress-theme-prose/components/Hero.vue (hunk 4 above)
Comment (ishan-krishna-okta) [reply packages/@okta/vuepress-theme-prose/components/Hero.vue]: Oops.. forgot to turn this flag off. Thank you for noticing. 🤦

**Review by emikhasyak-okta — APPROVED** [review]
Great job! 🚢

**Pull request merged**